kong/api/routes/server_names.lua

```diff
@@ -0,0 +1,16 @@
+local function method_not_allowed(self, db, helpers)
+  return helpers.responses.send_HTTP_METHOD_NOT_ALLOWED()
+end
+
+return {
+  -- GET / PATCH / DELETE /server_names/server_name are the only methods allowed
+
+  ["/server_names"] = {
+    before = method_not_allowed,
```

Comment:
If they are not allowed, is `404` a better one? E.g. is there any endpoint `/server_names` that you can call with any method? If not with any, then I would go with `404`. We could also add some support for autogenerator to remove endpoints, but not in this PR.